### ISSUE TITLE
t.w2.twscgi.SCGIClientProtocol causes an exception to be raised in t.w2.log._LogByteCounter

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1755]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#1755|
|Type|defect|
|Created|2006-05-26 21:54:18Z|

t.w2.twscgi.SCGIClientProtocol assumes that the http.Response object will not be mutated after it is returned.  This is not a valid assumption the response may be mutated by other systems such as t.w2.log.  SCGIClientProtocol, should instead keep a direct reference to the stream it returns in the outgoing response and write to that, as the response.stream is changed by t.w2.log for the purposes of byte counting.

This bug brought to you by the letter A.

<details><summary>Searchable metadata</summary>

```
trac-id__1755 1755
type__defect defect
reporter__dreid dreid
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1148680458000000 1148680458000000
changetime__1149873032000000 1149873032000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

branched to [branches/twscgi-stream-fix-1755](branches/twscgi-stream-fix-1755)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @jerub<br>@dreid set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @dreid|
|-|-|

[20060607T150006]< Jerub> dreid: my comments were something like, "comment on why self.response is set to 
                          None but self.stream isn't" and merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `closed`|
|-|-|

(In [17135]) Make twscgi not rely on self.response not being mutated after it is returned.  In certain situations such as logging the response is mutated to wrap the stream object, this causes an issue with doing self.response.stream.write("Foo") after you have returned self.response because self.response.stream is not the object you thought it was.

Author: dreid
Reviewer: Jerub
Fixes [#1755](https://github.com/twisted/twisted/issues/1755)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

Reverted at r17135 - the branch uses Python 2.4 syntax, but Python 2.3 is a supported platform.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set status to `new`|
|-|-|

Nasty bug. Thanks exarkun for that revert. Please merge now dreid.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `closed`|
|-|-|

(In [17142]) Fix twscgi to write to the stream directly instead of to the stream on the response object it has returned.

Author: dreid
Reviewer: Jerub
Fixes [#1755](https://github.com/twisted/twisted/issues/1755)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
